### ISSUE TITLE
EDSC-1168: EDSC Automatic Download Script replaces >> with &gt;&gt; when downloaded.

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#71]
Automatic Download Script replaces > with &amp;gt; where it shouldn't.  
For example, line 20

       echo "\nmachine urs.earthdata.nasa.gov\tlogin $username\tpassword $password" &amp;gt;&amp;gt; $netrc

should be

        echo "\nmachine urs.earthdata.nasa.gov\tlogin $username\tpassword $password" >> $netrc

Note that the generated HTML version looks okay, this only happens when you "download" the script. Workaround for now is to copy and paste the script from the website.

**Comment by earthdata-tasktop** [comment]
(Comment from Matthew Crouch):

[~amy.steiker] The comments on EDSC-1173 sound like what you are describing. [~bo.sun] Does that sound right to you?

**Comment by earthdata-tasktop** [comment]
(Comment from Amy Steiker):

Tested in UAT: Downloaded script from GLAH14 order ([https://search.uat.earthdata.nasa.gov/data/retrieve/3903625045](https://search.uat.earthdata.nasa.gov/data/retrieve/3903625045)). The script ran with the correct ">>" characters; however I received an error:

awk: can't open file /Users/asteiker/.netrc  

 source line number 1

The last granule was not downloaded; all others were downloaded successfully. I also tried this with a MODIS order ([https://search.uat.earthdata.nasa.gov/data/retrieve/5637329640](https://search.uat.earthdata.nasa.gov/data/retrieve/5637329640)). I received the same error, but this time all granules were successfully downloaded. Scripts are attached. I can verify that the characters are now fixed but it does look like there is a separate issue with this error and missing granule.  
[^EDSC-1168_download_MOD10A1.sh]  [^EDSC-1168_download_GLAH14.sh]

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

This was deployed with v1.28.6. [~ccuadrado] can verify this issue.

**Comment by earthdata-tasktop** [comment]
(Comment from Matthew Crouch):

[~amy.steiker] The comments on EDSC-1173 sound like what you are describing. [~bo.sun] Does that sound right to you?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[https://bugs.earthdata.nasa.gov/browse/EDSC-1215](https://bugs.earthdata.nasa.gov/browse/EDSC-1215) was approved to be worked this morning if you guys want to keep an eye on resolving that error.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~macrouch] and [~amy.steiker], although I couldn't download any granules when I ran the script, I did see the "awk" error. However, it is not related to either EDSC-1168 or EDSC-1173 but a separate one that happens when the script is done running and starts to clean up .netrc file. This is also reproduceable in production. I will file a ticket for it. EDSC-1215 is filed.

As to why I couldn't download the granules, I think maybe the server is behind a firewall/vpn? [~macrouch] were you able to get the granules using the script? I would also like to see the complete output after you ran the GLAH14 script. I couldn't figure out why the last granule was not downloaded.

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

This was deployed with v1.28.6. [~ccuadrado] can verify this issue.

**Comment by earthdata-tasktop** [comment]
(Comment from Matthew Crouch):

[~amy.steiker] The comments on EDSC-1173 sound like what you are describing. [~bo.sun] Does that sound right to you?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[https://bugs.earthdata.nasa.gov/browse/EDSC-1215](https://bugs.earthdata.nasa.gov/browse/EDSC-1215) was approved to be worked this morning if you guys want to keep an eye on resolving that error.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~macrouch] and [~amy.steiker], although I couldn't download any granules when I ran the script, I did see the "awk" error. However, it is not related to either EDSC-1168 or EDSC-1173 but a separate one that happens when the script is done running and starts to clean up .netrc file. This is also reproduceable in production. I will file a ticket for it. EDSC-1215 is filed.

As to why I couldn't download the granules, I think maybe the server is behind a firewall/vpn? [~macrouch] were you able to get the granules using the script? I would also like to see the complete output after you ran the GLAH14 script. I couldn't figure out why the last granule was not downloaded.

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[https://bugs.earthdata.nasa.gov/browse/EDSC-1215](https://bugs.earthdata.nasa.gov/browse/EDSC-1215) was approved to be worked this morning if you guys want to keep an eye on resolving that error.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~macrouch] and [~amy.steiker], although I couldn't download any granules when I ran the script, I did see the "awk" error. However, it is not related to either EDSC-1168 or EDSC-1173 but a separate one that happens when the script is done running and starts to clean up .netrc file. This is also reproduceable in production. I will file a ticket for it. EDSC-1215 is filed.

As to why I couldn't download the granules, I think maybe the server is behind a firewall/vpn? [~macrouch] were you able to get the granules using the script? I would also like to see the complete output after you ran the GLAH14 script. I couldn't figure out why the last granule was not downloaded.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~macrouch] and [~amy.steiker], although I couldn't download any granules when I ran the script, I did see the "awk" error. However, it is not related to either EDSC-1168 or EDSC-1173 but a separate one that happens when the script is done running and starts to clean up .netrc file. This is also reproduceable in production. I will file a ticket for it. EDSC-1215 is filed.

As to why I couldn't download the granules, I think maybe the server is behind a firewall/vpn? [~macrouch] were you able to get the granules using the script? I would also like to see the complete output after you ran the GLAH14 script. I couldn't figure out why the last granule was not downloaded.
